### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

**What this PR does / why we need it**:
I feel the README should be clearer on _what_ this repo contains. The first line indicates "This repository provides tools and scripts for building and testing", which is true but not it's main purpose. This PR also:
* removes the statement that the vSphere cloud provider is not production ready, wtih v1.0.0 we are now recommending this in production. 
* adds deprecation note about the in-tree cloud provider
* adds documentation section

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+# Kubernetes vSphere Cloud Provider
 
-## Cloud controller manager for vSphere
+## vSphere Cloud Controller Manager
 
-This repository provides tools and scripts for building and testing `Kubernetes cloud-controller-manager` for vSphere. The project is under development and should not be used in production.
+This repository contains the [Kubernetes cloud-controller-manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/) for vSphere.
 
-The in-tree vSphere cloud provider code is located within the [Kubernetes repository](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/legacy-cloud-providers/vsphere). If you want to create issues or pull requests for the in-tree cloud provider, please go to the [Kubernetes repository](https://github.com/kubernetes/kubernetes).
+This project is replaces the deprecated in-tree vSphere cloud provider located within the [Kubernetes repository](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/legacy-cloud-providers/vsphere). If you want to create issues or pull requests for the in-tree cloud provider, please go to the [Kubernetes repository](https://github.com/kubernetes/kubernetes).
```
Comment (codenrhoden) [inline README.md]: s/is//

Comment (andrewsykim) [reply README.md]: fixed

**Pull request merged**